Basic event selection

Fnux opened the pull request:
This PR address #2:
  * Replaces the default MainActivity by EventPickingActivity
  * List and allow select the existing events

Commit:
new event activity-> 3 buttons for general information

Commit:
add 3 activities map, schedule and ticketActivity and link buttons to activities

Fnux committed:
Create the EventPicking Activity

Fnux committed:
Create the EventPicking Activity

Fnux committed:
Display a simple list on the EventPickingActivity

Fnux committed:
Event list: add padding and a selection button to the items

Fnux committed:
Event list: replace currently useless toolbar and floating button by
simple help text

Fnux committed:
Delete "MainActivity", set the default activity to EventPickingActivity

Fnux committed:
Delete "MainActivity", set the default activity to EventPickingActivity

Fnux committed:
Remove a few obvious comments imported from android's documentation

zyuiop committed:
Create a class to represent an event

This class only holds data, and will not be used to directly query stuff. Instead, it will be passed to the backend repository when things need to be queried.

zyuiop committed:
Rename the "backend" package to "repository"

Somehow follows the recommended app architecture guidelines: https://developer.android.com/jetpack/docs/guide#fetch-data

zyuiop committed:
Add a new class representing an event organizer

zyuiop committed:
Create a first version of the EventRepository

This still doesn't hold a backend service, as we don't yet know how the backend will be handled. It however returns "LiveData" objects, which I think are a good way to handle things.

zyuiop committed:
Add the Dagger2 library for dependency injection

Dependency injection is a recommended pattern for providing Repositories to Models by the Android Jetpack good practices, described here https://developer.android.com/jetpack/docs/guide#overview

zyuiop committed:
Add dependency-injection logic to connect the backend with the view

Use Dagger2 for dependency injection stuff. Upgrade java to support lambda expressions.

zyuiop committed:
Add dependency-injection logic to connect the backend with the view

Use Dagger2 for dependency injection stuff. Upgrade java to support lambda expressions.

Fnux committed:
Move event-specific activities under the ui.eventShowcase, add intent to
selectEvent button

Fnux committed:
EventSelector: replace button by click on list item, pass event ID

Fnux committed:
Re-import small file lost due to human error in last rebase

Fnux committed:
Event Selector: add divider and increase spacing between items

Fnux committed:
Event Showcase: "Handle" invalid event IDs passed from previous activity

Fnux committed:
Travis CI: fix Travis configuration file (broken YAML indentation)

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventListAdapter.java:
```diff
@@ -0,0 +1,67 @@
+            Context context = itemView.getContext();
+            Intent intent = new Intent(context, EventActivity.class);
+            Event selectedEvent = mEvents.get(getAdapterPosition());
+            intent.putExtra(EventPickingActivity.SELECTED_EVENT_ID, selectedEvent.getId());
```
Don't we want to pass directly the Event for now?

Fnux replied:
We currently get the event back from the ID in the `ui.eventShowcase.EventActivity` activity: I think it's enough for the current sprint, but we indeed have to think harder for the next sprint.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/EventActivity.java:
```diff
@@ -0,0 +1,46 @@
+        setContentView(R.layout.activity_event);
+
+        Intent intent = getIntent();
+        int eventID = intent.getIntExtra(EventPickingActivity.SELECTED_EVENT_ID, -1);
```
Same thing, it would be easier if we directly got an Event from the Intent

zyuiop committed:
Set up dependency injection for the new activities

zyuiop committed:
Rename some classes to avoid confusion

The EventPickingActivityModule was renamed EventPickingModule, as the module contains more than just an activity.
The EventListModel was renamed to EventPickingModel, as it is a model for the whole module, and might do more than just fetching the list in the future.
The DaggerViewModelFactory was renamed ViewModelFactory to avoid confusion with Dagger generated classes that are prefixed by the Dagger name.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/viewmodel/ViewModelFactory.java:
```diff
@@ -0,0 +1,44 @@
+        this.creators = creators;
+    }
+
+    @SuppressWarnings("unchecked")
```
What should we do about this?

Fnux replied:
I don't see any other way to express it? We might exempt this method from the CognitiveComplexity check or increase the general complecity allowed - the check seem touchy -.

zyuiop replied:
We should exclude only this method for now, and we'll see later if this check comes out too often

Fnux committed:
Event Activity: display description of event passed from selector

Fnux commented on app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/Event.java:
```diff
@@ -0,0 +1,60 @@
+import android.graphics.Bitmap;
+
+/**
+ * This class holds the basic elements about an organized event.<br>
```
An HTML tag crept into this comment!

zyuiop replied:
Indeed, JavaDoc can hold HTML tags, especially to force linebreaks

Fnux committed:
Make layout IDs naming coherent, remove rogue HTML tag from comment

Fnux committed:
Revert "remove rogue HTML tag from comment"
